metaflow/datastore/s3_backend.py

```diff
@@ -220,26 +204,23 @@ def load_bytes(self, paths):
         if len(paths) == 0:
             return {}
 
-        self.reset_datatools_client()
+        s3 = S3(s3root=self.datastore_root,
+                tmproot=os.getcwd(), external_client=self.s3_client)
+        to_return = S3BytesLoader(s3)
         # We similarly do things in parallel for many files. This is again
         # a hack.
-        to_return = {}
         if len(paths) > 10:
-            results = self.s3_datatool.get_many(
-                paths, return_missing=True, return_info=True)
+            results = s3.get_many(paths, return_missing=True, return_info=True)
             for r in results:
                 if r.exists:
-                    to_return[r.key] = (FileWithClientRef(
-                        self.s3_datatool, r.path), r.metadata)
+                    to_return[r.key] = (LazyFile(r.path), r.metadata)
```

Comment:
why do we need `LazyFile` here instead of just passing `r.path` (no FD).

It seems you just open and read files sequentially here https://github.com/Netflix/metaflow/blob/alternative2-s3-cleanup/metaflow/datastore/task_datastore.py#L744 and here https://github.com/Netflix/metaflow/blob/alternative2-s3-cleanup/metaflow/datastore/flow_datastore.py#L158 so you shouldn't use any more FDs even without `LazyFile`

Reply:
I didn't want to make it so that it always had to be a file so I don't want to return a path. I could return something else though that has a read interface.

Reply:
For reference, we moved to a file path for now.